### PR TITLE
fix: harden type safety for analyzer/cache and unify kana character types

### PR DESCRIPTION
**Pull request by LaiTszKin**
## Related Issues / Motivation
- Related: none
- This PR addresses three high-impact type-safety hotspots that could bypass compile-time guarantees and lead to runtime failures in user-facing flows:
  - Japanese text analysis API relied on a private `_analyzer` field plus unsafe casting.
  - Kanji/Vocabulary cache stores used `undefined as Storage` casting in non-browser contexts.
  - Kana challenge modes had duplicated/inconsistent `KanaCharacter` definitions and required double casts.

## Engineering Decisions and Rationale
- Replaced private-field access in analyze-text API with a dedicated `KuromojiAnalyzer` singleton (`init` + `parse`) and explicit retry-safe initialization.
  - Rationale: avoid coupling to undocumented internals and keep initialization failure recoverable.
- Removed SSR-unsafe storage casting in persisted stores by using `storage: undefined` when `window` is unavailable.
  - Rationale: preserve runtime safety without lying to the type system.
- Unified kana type source to `flattenKanaGroup` and removed `as unknown as` casts in Blitz/Gauntlet.
  - Rationale: keep one canonical shape for `KanaCharacter`, restoring meaningful type checks.
- Added API route tests for:
  - invalid payload handling,
  - token normalization path,
  - analyzer init failure followed by successful retry.
  - Rationale: encode edge-case behavior to prevent regressions.

## Test Results and Commands
- ✅ `npx tsc --noEmit --pretty false`
- ✅ `npx eslint app/api/analyze-text/route.ts app/api/analyze-text/route.test.ts features/Vocabulary/store/useVocabCacheStore.ts features/Kanji/store/useKanjiCacheStore.ts features/Kana/lib/generateKanaQuestions.ts features/Kana/components/Blitz.tsx features/Kana/components/Gauntlet.tsx`
- ✅ `npx vitest run app/api/analyze-text/route.test.ts`
- All listed checks passed locally.

### Test Cases (for complex changes)
- Case 1: empty `text` payload -> returns `400 INVALID_INPUT`.
- Case 2: analyzer returns kuromoji token with katakana reading -> response contains normalized hiragana reading and mapped POS.
- Case 3: analyzer `init()` fails once -> first request returns 500, next request retries initialization and succeeds.

### CODE REVIEW

File: app/api/analyze-text/route.ts [hunk 1]
```diff
@@ -76,35 +73,45 @@ function cleanupCache() {
...
-
-  const kuroshiro = new Kuroshiro();
-  const analyzer = new KuromojiAnalyzer();
-  await kuroshiro.init(analyzer);
-  kuroshiroInstance = kuroshiro as unknown as KuroshiroInstance;
-  return kuroshiroInstance;
+  kuromojiAnalyzerInitPromise = (async () => {
+    const { default: KuromojiAnalyzer } =
+      await import('kuroshiro-analyzer-kuromoji');
+    const analyzer = new (KuromojiAnalyzer as KuromojiAnalyzerConstructor)();
```
[the remark this answers is not in the document]
Comment (LaiTszKin) [reply app/api/analyze-text/route.ts]: 已依建議更新：在 kuroshiro.d.ts 補上 parse(text): Promise<KuromojiToken[]> 與 KuromojiToken 型別，並移除 route 中的 constructor type cast。變更已在 commit 8f421525 推送。
